File: __Viejo/Formularios/Busqueda/frm_ultimo_comprobante.vb
Public Class frm_ultimo_comprobante
    Private Sub frm_ultimo_comprobante_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        cargar_combo(cmb_comprobante, "SELECT id_comprobante, comprobante FROM comprobantes WHERE activo = '1' AND esElectronica = '1' ORDER BY comprobante ASC", basedb, "comprobante", "id_comprobante")
        cmb_comprobante.Text = "Seleccione un comprobante..."
    End Sub

    Private Sub cmb_comprobante_KeyPress(sender As Object, e As KeyPressEventArgs) Handles cmb_comprobante.KeyPress
        'e.KeyChar = ""
    End Sub

    Private Sub cmd_ok_Click(sender As Object, e As EventArgs) Handles cmd_ok.Click
        If cmb_comprobante.Text = "Seleccione un comprobante..." Then
            MsgBox("Debe seleccionar un comprobante para ejecutar la consulta", vbExclamation + vbOKOnly, "Centrex")
            Exit Sub
        End If

        Dim c As New comprobante
        Dim ultimoComprobante As Integer
        c = info_comprobante(cmb_comprobante.SelectedValue)

        ' NO usar Try-Catch aquí para permitir que las excepciones se propaguen
        ' y se muestre el formulario de errores detallados desde WSFEv1 o WSAA
        ultimoComprobante = consultaUltimoComprobante(c.puntoVenta, c.id_tipoComprobante, c.numeroComprobante)

        MsgBox("El último número de comprobante para el documento: " + c.comprobante + " con el punto de venta: " + c.puntoVenta.ToString +
            " es: " + ultimoComprobante.ToString, vbInformation + vbOKOnly, "Centrex")
    End Sub

    Private Sub frm_ultimo_comprobante_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        closeandupdate(Me)
    End Sub
End Class

File: Formularios/Busqueda/frm_ultimo_comprobante.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace Centrex
{
    public partial class frm_ultimo_comprobante
    {
        public frm_ultimo_comprobante()
        {
            InitializeComponent();
        }
        private void frm_ultimo_comprobante_Load(object sender, EventArgs e)
        {
            var filtros = new Dictionary<string, object>
            {
                ["EsElectronica"] = true
            };
            var orden = new List<Tuple<string, bool>> { Tuple.Create("Comprobante", true) };
            generales.Cargar_Combo(
                ref cmb_comprobante,
                entidad: "ComprobanteEntity",
                displaymember: "Comprobante",
                valuemember: "IdComprobante",
                predet: -1,
                soloActivos: true,
                filtros: filtros,
                orden: orden);
            cmb_comprobante.SelectedIndex = -1;
        }

        private void cmb_comprobante_KeyPress(object sender, KeyPressEventArgs e)
        {
            // e.KeyChar = ""
        }

        private void cmd_ok_Click(object sender, EventArgs e)
        {
            if (cmb_comprobante.SelectedValue is null)
            {
                Interaction.MsgBox("Debe seleccionar un comprobante para ejecutar la consulta", (MsgBoxStyle)((int)Constants.vbExclamation + (int)Constants.vbOKOnly), "Centrex");
                return;
            }

            var c = new ComprobanteEntity();
            int ultimoComprobante;
            c = comprobantes.info_comprobante(Conversions.ToInteger(cmb_comprobante.SelectedValue));

            // NO usar Try-Catch aquí para permitir que las excepciones se propaguen
            // y se muestre el formulario de errores detallados desde WSFEv1 o WSAA
            ultimoComprobante = factura_electronica.ConsultaUltimoComprobante(c.PuntoVenta, c.IdTipoComprobante, c.NumeroComprobante, c.Testing);

            Interaction.MsgBox("El último número de comprobante para el documento: " + c.Comprobante + " con el punto de venta: " + c.PuntoVenta.ToString() + " es: " + ultimoComprobante.ToString(), (MsgBoxStyle)((int)Constants.vbInformation + (int)Constants.vbOKOnly), "Centrex");
        }

        private void frm_ultimo_comprobante_FormClosed(object sender, FormClosedEventArgs e)
        {
            closeandupdate(this);
        }
    }
}
